exporters/otlp/common/src/main/java/io/opentelemetry/exporter/otlp/internal/okhttp/OkHttpExporter.java

```diff
@@ -94,6 +94,7 @@ public void onFailure(Call call, IOException e) {
               @Override
               public void onResponse(Call call, Response response) {
                 if (response.isSuccessful()) {
+                  response.close();
```

Comment:
We only close the response when `response.isSuccessful()`?

How about we change this to
```
  @Override
  public void onResponse(Call call, Response response) {
    try (ResponseBody responseBody = response.body()) {
      if (response.isSuccessful()) {
        exporterMetrics.addSuccess(numItems);
        result.succeed();
        return;
      }

      exporterMetrics.addFailed(numItems);
      int code = response.code();

      String status = extractErrorStatus(responseBody, response.message());

      logger.log(
          Level.WARNING,
          "Failed to export "
              + type
              + "s. Server responded with HTTP status code "
              + code
              + ". Error message: "
              + status);
      result.fail();
    }
  }
});
```
And `extractErrorStatus` to:
```
private static String extractErrorStatus(
    @Nullable ResponseBody responseBody, String responseMessage) {
  if (responseBody == null) {
    return "Response body missing, HTTP status message: " + responseMessage;
  }
  try {
    return GrpcStatusUtil.getStatusMessage(responseBody.bytes());
  } catch (IOException e) {
    return "Unable to parse response body, HTTP status message: " + responseMessage;
  }
}
```

Reply:
Yes, agree with @jack-berg . This is more idiomatic okhttp usage.

Reply:
Yeah, the autocloseable is nice here. I had to look up/refresh what the behavior is for nullable objects in autocloseable usage, but it's null safe.  Passing that nullable around tho is a bit gross, but the guard was already in `extractErrorStatus()` so not a biggie.

There are some comments in the docs for `close()` about it being an error to close() some types of response, maybe that was the reason? Anyway, those cases (cache response, prior redirect response) don't seem relevant here anyway....so I'd +1 the autocloseable.

Reply:
For reference, the error case was closing implicitly through the call to `.bytes()`. But agree the try resources is clearer